bluemira/base/reactor.py

```diff
@@ -446,26 +551,64 @@ def show_cad(
         filter_:
             A callable to filter Components from the Component tree,
             returning True keeps the node False removes it
+        filename:
+            the filename to save, will default to the component name
+        formatt:
+            CAD file format
+        directory:
+            Directory to save into
+        kwargs:
+            passed to the `~bluemira.geometry.tools.save_cad` function
         """
-        dims_to_show = self._validate_cad_dims(*dims, **kwargs)
+        if filename is None:
+            filename = self.name
 
-        # We filter because self.component (above) only creates
-        # a new root node for this reactor, not a new component tree.
-        comp_copy = self._filter_tree(
-            self.component(with_components), dims_to_show, filter_
+        save_cad(
+            self._filter_and_reconstruct(
+                self._validate_cad_dims(*dims), with_components, n_sectors, filter_
+            ),
+            Path(directory, filename).as_posix(),
+            formatt,
+            **kwargs,
         )
-        # if "xyz" is requested, construct the 3d cad
-        # from each xyz component in the tree,
-        # as it's assumed that the cad is only built for 1 sector
-        # and is sector symmetric, therefore can be patterned
-        if "xyz" in dims_to_show:
-            self._construct_xyz_cad(
-                comp_copy,
-                with_components,
-                self.n_sectors if n_sectors is None else n_sectors,
-            )
 
-        ComponentDisplayer().show_cad(comp_copy, **kwargs)
+    def show_cad(
+        self,
+        *dims: str,
+        with_components: Optional[List[ComponentManager]] = None,
+        n_sectors: Optional[int] = None,
+        filter_: Union[Callable[[Component], bool], None] = FilterMaterial(),
+        **kwargs,
+    ):
+        """
+        Show the CAD build of the reactor.
+
+        Parameters
+        ----------
+        *dims:
+            The dimension of the reactor to show, typically one of
+            'xz', 'xy', or 'xyz'. (default: 'xyz')
+        with_components:
+            The components to construct when displaying CAD for xyz.
+            Defaults to None, which means show "all" components.
+        n_sectors:
+            The number of sectors to construct when displaying CAD for xyz
+            Defaults to None, which means show "all" sectors.
+        filter_:
+            A callable to filter Components from the Component tree,
+            returning True keeps the node False removes it
+        kwargs:
+            passed to the `~bluemira.display.displayer.show_cad` function
+        """
+        ComponentDisplayer().show_cad(
```

Comment:
I'm getting this error when running EUDEMO

Reply:
![image](https://github.com/Fusion-Power-Plant-Framework/bluemira/assets/3520445/c1032f74-64d2-42d2-b371-42e6ad84dca2)

Reply:
good catch see #2289